### PR TITLE
chore: run unit test with matrix strategy

### LINKED ISSUE #5: Enhance the test matrix
Please refer to [tests.yaml#L36-L41](https://github.com/etcd-io/etcd/blob/6a156bd5556be3fd562b4eb20eed0e5e1f34bcc5/.github/workflows/tests.yaml#L36-L41) to enhance the test workflow for raft

### PR DESCRIPTION
**Pull request by nic-chen**
closes #5

### CODE REVIEW

**Comment by nic-chen** [comment]
cc @ahrtr have a review, thanks.

File: scripts/test.sh [hunk 1]
```diff
@@ -4,6 +4,23 @@ set -eo pipefail
 
 source ./scripts/test_lib.sh
 
+# determine whether target supports race detection
```
Comment (ahrtr) [inline scripts/test.sh:7]: PLease also add,

```
if [ -z "$GOARCH" ]; then
  GOARCH=$(go env GOARCH);
fi
```

**Comment by ahrtr** [comment]
Thanks @nic-chen for working on this. 

I see that the second commit is "Merged main into ...". Please always rebase PR instead of merging main into your branch.

File: scripts/test.sh (hunk 1 above)
Comment (nic-chen) [reply scripts/test.sh:12]: added.

**Comment by nic-chen** [comment]
> Thanks @nic-chen for working on this.
> 
> I see that the second commit is "Merged main into ...". Please always rebase PR instead of merging main into your branch.

I am used to not squashing commits in pr, but squashing when merging the pr ...

Thanks for the reminder, I will pay attention to it next time.

**Comment by ahrtr** [comment]
The PR looks good to me. The only concern is the second commit [Merge branch 'main' into ci/unit-test-matrix](https://github.com/etcd-io/raft/pull/7/commits/a23c946f3c6dd59a291102cd828adcec48ba3ebb) doesn't make any sense. Please consider to squash the 2nd and 3rd commit into one. The command is roughly something like below,

```
git rebase -i HEAD~2
```

**Comment by nic-chen** [comment]
> git rebase -i HEAD~2

I've tried it before and it will include commits from the main branch, I'm not sure if it will affect the main branch.

image

Maybe I can commit to another branch if needed ？

**Comment by nic-chen** [comment]
@ahrtr  updated.

**Review by ahrtr — APPROVED** [review]
LGTM 

Thank you @nic-chen 

One minor comment is there are some noises in the commit message. It's easy to cleanup them. You just need to execute "`git commit --amend`", and then cleanup all the noises and keep only the message which is useful.  Please feel free to fix it if you want.

**Comment by nic-chen** [comment]
> LGTM
> 
> Thank you @nic-chen
> 
> One minor comment is there are some noises in the commit message. It's easy to cleanup them. You just need to execute "`git commit --amend`", and then cleanup all the noises and keep only the message which is useful. Please feel free to fix it if you want.

fixed. thanks!

**Pull request merged**